crates/rspack_fs/src/write.rs

```diff
@@ -40,3 +40,81 @@ pub trait WritableFileSystem: Debug + Send + Sync {
 
   fn stat<'a>(&'a self, file: &'a Utf8Path) -> BoxFuture<'a, Result<FileMetadata>>;
 }
+
+/// Extension trait for [`AsyncWritableFileSystem`].
+pub trait WritableFileSystemExt: WritableFileSystem {
```

Comment:
Don't think the file system should be modified. Perhaps an utility function should be added, and the file system is used inside to handle this.